### PR TITLE
DLS1: Implement the 'count' action

### PR DESCRIPTION
**Pull request by MikeIsAStar**

### CODE REVIEW

File: nhttp/ascii.go [hunk 1]
```diff
@@ -66,6 +66,25 @@ func lower(b byte) byte {
 	return b
 }
 
+// Checks if the given string is composed exclusively of uppercase alphanumeric characters.
+func IsUppercaseAlphanumeric(str string) bool {
+	len := len(str)
```
Comment (noahpistilli) [inline nhttp/ascii.go:71]: You shouldn't overshadow the function `len` with a variable of the same name

File: nhttp/ascii.go [hunk 2]
```diff
@@ -66,6 +66,25 @@ func lower(b byte) byte {
 	return b
 }
 
+// Checks if the given string is composed exclusively of uppercase alphanumeric characters.
+func IsUppercaseAlphanumeric(str string) bool {
```
Comment (noahpistilli) [inline nhttp/ascii.go:70]: This function should probably go in `common/strings.go`. This directory is a modified version of Go's HTTP library.

**Review by noahpistilli — CHANGES_REQUESTED** [review]
Other than the requested changes, it looks good.

I don't like how NAS's handling of HTTP functions has turned into though. Will probably be refactored later.

File: nhttp/ascii.go (hunk 2 above)
Comment (MikeIsAStar) [reply nhttp/ascii.go]: Resolved in commit [333c797](https://github.com/WiiLink24/wfc-server/pull/20/commits/333c79701d2a04603be12dd7852679fcd4353729).

**Comment by MikeIsAStar** [comment]
> I don't like how NAS's handling of HTTP functions has turned into though.

Agreed.

**Comment by mkwcat** [comment]
> I don't like how NAS's handling of HTTP functions has turned into though. Will probably be refactored later.
I had to rewrite it because the old system was specifically designed for the `/ac` endpoint and didn't work with anything else. I don't see what's so wrong with it though, imo it's a lot easier to understand at a glance than how it was before

**Review by noahpistilli — APPROVED** [review]

**Comment by noahpistilli** [comment]
> I had to rewrite it because the old system was specifically designed for the /ac endpoint and didn't work with anything else. 

By adding a new handle group for each route `pr`, `download`, `payload`, `online`, you could have separate commands handle each. 

Disregarding that, I don't like that PR and download are handled inside of `handleAuthRequest`

**Comment by mkwcat** [comment]
> By adding a new handle group for each route `pr`, `download`, `payload`, `online`, you could have separate commands handle each.

No that's not how it worked, a "group" was just a list of actions defined in the `action` field, but `ac` and `download` are the only endpoints/groups that handle actions like that, so it had to be rewritten regardless to support more endpoints

**Pull request merged**